sentinel-extension/sentinel-datasource-extension/src/main/java/com/alibaba/csp/sentinel/datasource/FileWritableDataSource.java

```diff
@@ -40,18 +41,25 @@
 
     private final Lock lock = new ReentrantLock(true);
 
-    public FileWritableDataSource(String filePath, Converter<T, String> configEncoder) {
+    public FileWritableDataSource(String filePath, Converter<T, String> configEncoder) throws IOException {
         this(new File(filePath), configEncoder);
     }
 
-    public FileWritableDataSource(File file, Converter<T, String> configEncoder) {
+    public FileWritableDataSource(File file, Converter<T, String> configEncoder) throws IOException {
         this(file, configEncoder, DEFAULT_CHARSET);
     }
 
-    public FileWritableDataSource(File file, Converter<T, String> configEncoder, Charset charset) {
+    public FileWritableDataSource(File file, Converter<T, String> configEncoder, Charset charset) throws IOException {
         if (file == null || file.isDirectory()) {
             throw new IllegalArgumentException("Bad file");
         }
+        if (!file.exists()) {
```

Comment:
It's better to abstract the create-if-not-exist code snippet as a method and add a test case for it.

Reply:
Friendly ping :)